api/src/test/java/com/epam/pipeline/acl/security/AclPermissionApiServiceTest.java

```diff
@@ -0,0 +1,272 @@
+                .isEqualTo(aclSecuredEntry);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyDeleteAllPermissionsForNotOwner() {
+        doReturn(s3bucket).when(entityManager).load(AclClass.DATA_STORAGE, ID);
+        mockUser(ANOTHER_SIMPLE_USER);
+
+        assertThrows(AccessDeniedException.class,
+            () -> aclPermissionApiService.deleteAllPermissions(ID, AclClass.DATA_STORAGE));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldChangeOwnerForAdmin() {
+        doReturn(aclSecuredEntry).when(spyPermissionManager)
+                .changeOwner(ID, AclClass.DATA_STORAGE, SIMPLE_USER);
+
+        assertThat(aclPermissionApiService.changeOwner(ID, AclClass.DATA_STORAGE, SIMPLE_USER))
+                .isEqualTo(aclSecuredEntry);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldChangeOwnerForOwner() {
+        doReturn(s3bucket).when(entityManager).load(AclClass.DATA_STORAGE, ID);
+        doReturn(aclSecuredEntry).when(spyPermissionManager)
+                .changeOwner(ID, AclClass.DATA_STORAGE, SIMPLE_USER);
+        mockUser(SIMPLE_USER);
+
+        assertThat(aclPermissionApiService.changeOwner(ID, AclClass.DATA_STORAGE, SIMPLE_USER))
+                .isEqualTo(aclSecuredEntry);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyChangeOwnerForNotOwner() {
+        doReturn(s3bucket).when(entityManager).load(AclClass.DATA_STORAGE, ID);
+        mockUser(ANOTHER_SIMPLE_USER);
+
+        assertThrows(AccessDeniedException.class,
+            () -> aclPermissionApiService.changeOwner(ID, AclClass.DATA_STORAGE, SIMPLE_USER));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldLoadEntityPermissionForAdmin() {
+        doReturn(entityPermissionVO).when(spyPermissionManager)
+                .loadEntityPermission(AclClass.DATA_STORAGE, ID);
+
+        assertThat(aclPermissionApiService.loadEntityPermission(ID, AclClass.DATA_STORAGE))
+                .isEqualTo(entityPermissionVO);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadEntityPermissionForOwner() {
+        doReturn(s3bucket).when(entityManager).load(AclClass.DATA_STORAGE, ID);
+        doReturn(entityPermissionVO).when(spyPermissionManager)
+                .loadEntityPermission(AclClass.DATA_STORAGE, ID);
+        mockUser(SIMPLE_USER);
+
+        assertThat(aclPermissionApiService.loadEntityPermission(ID, AclClass.DATA_STORAGE))
+                .isEqualTo(entityPermissionVO);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadEntityPermissionForNotOwner() {
+        doReturn(s3bucket).when(entityManager).load(AclClass.DATA_STORAGE, ID);
+        mockUser(ANOTHER_SIMPLE_USER);
+
+        assertThrows(AccessDeniedException.class,
+            () -> aclPermissionApiService.loadEntityPermission(ID, AclClass.DATA_STORAGE));
+    }
+
+    private void mockUser(final String username) {
```

Comment:
This method has usages in another places (`ClusterApiServiceTest`, `ServerlessConfigurationApiServiceTest`, `AbstractDataStorageAclTest`). Maybe we should consider moving this method to `AbstractAclTest`?